src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +292,136 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
```

Comment:
where did the first `analyzer = mapper.fieldType().searchAnalyzer();` end up?

also, not sure it happens, but what happens if `parseContext.getSearchAnalyzer(mapper);` returns `null` ? the previous code falls back to the following case, while in your case you'd go ahead with that null value without fallback. Makes sense?